### DISCUSSION TITLE
Username is stored despite deleting account

### DISCUSSION THREAD

**Discussion by gsilvapt** [mastodon/mastodon#17488] [Q&A] [answered]
Hi everyone :wave:

I apologise before anything if this is the wrong place to ask these type of questions.

I was testing the application and noticed that old users do not get deleted after the user delete the account. I attest to that because when I tried registering back again, it said my old username was in use.

Is there a use case for this? Why would the service do this?

Thanks for your time!

**Comment by nightpool** [comment, marked as answer]
This is to prevent users from impersonating deleted accounts, and also means that we can reliably serve a 410 status to other applications to represent a deleted account, rather then a 404 status that is more ambiguous and might be the result of a transient error

**Reply by gsilvapt** [reply]
That's a good point, got it. Thanks!

**Comment by cham2019** [comment]
Hi. I'm new on Mastodon. It is a good point to keep an old username after deleting the account, to avoid possible impersonations in the future.  But what about the old posts of that  deleted local account?  Are old posts also removed, or are they kept in the local server occupying storage space indefinitely ?
